model/image.go

```diff
@@ -69,14 +70,158 @@ func (img *Image) GetSamples() []uint32 {
 // SetSamples convert samples to byte-data and sets for the image.
 func (img *Image) SetSamples(samples []uint32) {
 	resampled := sampling.ResampleUint32(samples, int(img.BitsPerComponent), 8)
-	var data []byte
-	for _, val := range resampled {
-		data = append(data, byte(val))
+	data := make([]byte, len(resampled))
+	for i, val := range resampled {
+		data[i] = byte(val)
 	}
 
 	img.Data = data
 }
 
+// ColorAt returns the color of the image pixel specified by the x and y coordinates.
+func (img *Image) ColorAt(x, y int) (gocolor.Color, error) {
```

Comment:
Can you add testcases to cover the different pathways/cases here?

Reply:
Sure.